test/Microsoft.ML.Tests/Scenarios/Api/AspirationalExamples.cs

```diff
@@ -29,30 +29,34 @@ public class IrisExample
         public void FirstExperienceWithML()
         {
             // Load the data into the system.
-            string dataPath = "iris-data.txt";
-            var data = TextReader.FitAndRead(env, dataPath, row => (
-                Label: row.ReadString(0),
-                SepalWidth: row.ReadFloat(1),
-                SepalLength: row.ReadFloat(2),
-                PetalWidth: row.ReadFloat(3),
-                PetalLength: row.ReadFloat(4)));
-
-
-            var preprocess = data.Schema.MakeEstimator(row => (
-                // Convert string label to key.
-                Label: row.Label.DictionarizeLabel(),
-                // Concatenate all features into a vector.
-                Features: row.SepalWidth.ConcatWith(row.SepalLength, row.PetalWidth, row.PetalLength)));
-
-            var pipeline = preprocess
-                // Append the trainer to the training pipeline.
-                .AppendEstimator(row => row.Label.PredictWithSdca(row.Features))
-                .AppendEstimator(row => row.PredictedLabel.KeyToValue());
-
-            // Train the model and make some predictions.
-            var model = pipeline.Fit<IrisExample, IrisPrediction>(data);
-
-            IrisPrediction prediction = model.Predict(new IrisExample
+            string dataPath = "iris.txt";
+            var dataReader = TextReader.CreateReader(env, dataPath, row => (
+                label: row.LoadText(0),
+                sepalWidth: row.LoadFloat(1),
+                sepalLength: row.LoadFloat(2),
+                petalWidth: row.LoadFloat(3),
+                petalLength: row.LoadFloat(4)));
+
+
+            var pipeline = dataReader.MakeEstimator()
+                .Append(row => (
+                    // Convert string label to key.
+                    label: row.Label.DictionarizeLabel(),
+                    // Concatenate all features into a vector.
+                    features: row.SepalWidth.ConcatWith(row.sepalLength, row.petalWidth, row.petalLength)))
+                .Append(row => row.label.PredictSdcaMultiClass(row.features))
+                .Append(row => row.predictedLabel.KeyToValue());
+
+            // Read the data
+            var data = reader.Read(dataPath);
+
+            // Fit the data
+            var model = estimator.Fit(data);
+
+            var predictor = model.MakePredictionFunction<IrisExample, IrisPrediction>();
```

Comment:
>MakePredictionFunction [](start = 34, length = 22)

@Zruty0, @tfinley@gmail.com what will this be called? is this the correct name?

Reply:
I see a CreateaPredictor on SDCA; that the naming convention?

---
In reply to: [215362963](https://github.com/dotnet/machinelearning/pull/812#discussion_r215362963) [](ancestors = 215362963)

Reply:
CreatePredictor is not public. We agreed to use 'Make' for public factory methods

---
In reply to: [215364904](https://github.com/dotnet/machinelearning/pull/812#discussion_r215364904) [](ancestors = 215364904,215362963)

Reply:
And yes, `MakePredictionFunction` is the correct name

---
In reply to: [215374117](https://github.com/dotnet/machinelearning/pull/812#discussion_r215374117) [](ancestors = 215374117,215364904,215362963)